keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
+    //        deleted from the OSP even if workloads are currently accessing 
+    //        this bucket
+    DeletionPolicy DeletionPolicy 
+    
+    // Name of the BucketClass specified in the BucketRequest
+    // +optional
+    BucketClassName string 
+    
+    // Name of the BucketRequest that resulted in the creation of this Bucket
+    // +optional
+    BucketRequest corev1.ObjectReference
+    
+    // AllowedNamespaces is the list of namespaces from which workloads  
+    // can access this Bucket. If empty, all namespaces can access this access
+    // +optional 
+    AllowedNamespaces []string 
+
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+    
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating the bucket
+    // +optional
+    Parameters map[string]string 
+  }    
+  
+  Status BucketStatus {
+    // BucketAvailable indicates that the bucket is ready for consumption
+    // by workloads
+    BucketAvailable bool
+    
+    // BucketID is the unique id for the bucket in the OSP
+    BucketID string
+  }
 }
 ```
 
-## ProvisonerCreateBucket
+## BucketRequest
 
-This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner. The call to *ProvisonerCreateBucket* MUST be idempotent.
+A request to create a Bucket in OSP. BucketRequest is a namespaced resource
 
+```yaml
+BucketRequest {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketRequestSpec {
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+    
+    // Name of the BucketClass 
+    // +optional    
+    BucketClassName string
+  }
+    
+  Status BucketRequestStatus {
+    // BucketAvailable indicates that the bucket is ready for consumpotion
+    // by workloads      
+    BucketAvailable bool 
+  }
 ```
-message ProvisionerCreateBucketRequest {    
-    // This field is REQUIRED
-    string bucket_name = 1;
+## BucketClass
 
-    map<string,string> bucket_context = 2;
+Resouce for configuring common properties for multiple Buckets. BucketClass is a clustered resource
 
-    enum AnonymousBucketAccessMode {
-	PRIVATE = 0;
-	PUBLIC_READ_ONLY = 1;
-	PUBLIC_WRITE_ONLY = 2;
-	PUBLIC_READ_WRITE = 3;
-    }
-    
-    AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
+```yaml
+BucketClass {
```

Comment:
Removing the protocol field from the BC implies, to me, that the protocol is baked into the driver (though it is also defined in the BR). That suggests that if a vendor supports >1 protocols then they need a separate driver for each.

Reply:
If the driver doesn't support the requested protocol, then ProvisionerCreateBucket gRPC call to the driver should fail with InvalidArguments